### PR TITLE
dbeaver/dbeaver#21676 datasource navigator  select null when mouse click on an empty space

### LINKED ISSUE #21676: datasource navigator donot select null when mouse click on an empty space
### Description

In the data source navigation bar on the left, when I selected a data source or folder and then clicked on a blank space, the system did not set the previously selected item to empty, which caused me to create a new data source or folder. under the selected folder.

otherwise,it works fine on  my workmate's mac os.

### DBeaver Version

23.2.3

### Operating System

windows 11

### Database and driver

any datasource in folder

### Steps to reproduce

1.create a folder.
2.select the folder.
3.click an empty space
4.create a new datasource (any type)
5.the datasouce doesnot show in root node, but show in the step2 folder instead
image

### Additional context

i found this problem during developing with dbeaver,and i have solved it 
image

how can i contribute it to community

### PR DESCRIPTION
**Pull request by itsyfx**

### CODE REVIEW

**Comment by E1izabeth** [comment]
Thank you for contribution

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/NavigatorViewBase.java [hunk 1]
```diff
@@ -135,6 +135,11 @@ public void mouseDown(MouseEvent e) {
             @Override
             public void mouseUp(MouseEvent e) {
                 super.mouseUp(e);
+                Point point = new Point(e.x, e.y);
+                TreeItem item = navigatorTree.getViewer().getTree().getItem(point);
+                if (item == null) {
+                    navigatorTree.getViewer().setSelection(new StructuredSelection());
+                }
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/NavigatorViewBase.java:145]: Why is this added only to mouseUp event handler?

Comment (itsyfx) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/NavigatorViewBase.java:145]: mouseUp or mouseDown event one is enough for sloving this problem. Its not necessary to add two events.

That is my idea😂.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/NavigatorViewBase.java:145]: Yeah, you're right

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
This PR can't be build
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:3.0.1:compile (default-compile) on project org.jkiss.dbeaver.ui.navigator: Compilation failure: Compilation failure: 
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce/workspace/dbeaver/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/NavigatorViewBase.java:[138] 
[ERROR] 	Point point = new Point(e.x, e.y);
[ERROR] 	^^^^^
[ERROR] Point cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce/workspace/dbeaver/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/NavigatorViewBase.java:[138] 
[ERROR] 	Point point = new Point(e.x, e.y);
[ERROR] 	                  ^^^^^
[ERROR] Point cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce/workspace/dbeaver/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/NavigatorViewBase.java:[139] 
[ERROR] 	TreeItem item = navigatorTree.getViewer().getTree().getItem(point);
[ERROR] 	^^^^^^^^
[ERROR] TreeItem cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce/workspace/dbeaver/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/NavigatorViewBase.java:[141] 
[ERROR] 	navigatorTree.getViewer().setSelection(new StructuredSelection());
[ERROR] 	                                           ^^^^^^^^^^^^^^^^^^^
[ERROR] StructuredSelection cannot be resolved to a type
```

**Comment by E1izabeth** [comment]
@itsyfx  can you build it in your local environment? The imports are missing.

**Comment by itsyfx** [comment]
> @itsyfx  can you build it in your local environment? The imports are missing.

 sorry ，i forgot that。 i will modify it next Monday。

after that，what should i do，PR again？

**Comment by E1izabeth** [comment]
Just make a new commit and push it in this branch (datasource_navigator_click_empty_fix). This PR will be automatically updated.

**Comment by itsyfx** [comment]
> Just make a new commit and push it in this branch (datasource_navigator_click_empty_fix). This PR will be automatically updated.

OK

**Comment by E1izabeth** [comment]
Passed to QA

**Comment by Matvey16** [comment]
Verified
